packages/amplify-e2e-tests/src/__tests__/smoketest.test.ts

```diff
@@ -0,0 +1,299 @@
+import execa from 'execa';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as pty from 'node-pty';
+import { nspawn as spawn } from '@aws-amplify/amplify-e2e-core';
+
+export const NPM = {
+  install(pkgName: string, isGlobal: boolean = false) {
+    const args = ['install'];
+    if (isGlobal) {
+      args.push('-g');
+    }
+    args.push(pkgName);
+    execa.sync('npm', args, { stdio: 'inherit' });
+  },
+  uninstall(pkgName: string, isGlobal: boolean = false) {
+    const args = ['uninstall', pkgName];
+    if (isGlobal) {
+      args.push('-g');
+    }
+    execa.sync('npm', args, { stdio: 'inherit' });
+  },
+};
+
+export type SmoketestArgs = {
+  projectDirectory: string;
+  cliVersion: string;
+  destructive: boolean;
+};
+
+function getArgs(): SmoketestArgs {
+  const { DESTRUCTIVE, CLI_VERSION, PROJECT_DIRECTORY } = process.env;
+  if (!CLI_VERSION) {
```

Comment:
I'd make CLI_VERSION an arg

Reply:
It was an arg previously, but I was having issues passing it in since the file is being run with jest. I didn't spend more than 10 minutes looking into it and just decided to switch to env vars. I can revisit this next week, as I personally preferred them as arguments.

Reply:
if passing args is hard then we should have `smoketest`, `smoketest-rc` , `smoketest-latest` targets. the latter two would set env var before calling jest.